### PR TITLE
Problem: ibc relayer don't work well with default tx prioritization strategy

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Change the default priority mechanism to be based on gas price

it'll be the default behavior in next SDK version: https://github.com/cosmos/cosmos-sdk/pull/12953

- [x] add integration test

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: app/validator_tx_fee.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+				return nil, 0, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
+			}
+		}
+	}
+
+	priority := getTxPriority(feeCoins, int64(gas))
+	return feeCoins, priority, nil
+}
+
+// getTxPriority returns a naive tx priority based on the amount of the smallest denomination of the fee
```
Comment (yihuang) [inline app/validator_tx_fee.go:46]:
```suggestion
// getTxPriority returns a naive tx priority based on the amount of the smallest denomination of the gas price
```

**Comment by tomtau** [comment]
should this one: https://github.com/crypto-org-chain/chain-main/pull/846 be merged first?

**Comment by yihuang** [comment]
> should this one: #846 be merged first?

no, prioritization don't need any change at relayer side, we just calculate the priority based on gas price rather than fee, relayer always use a static gas price.

**Review by tomtau — APPROVED** [review]
looks all right, the integration test may need to update pystarport or what the root cause for that failed transfer is

File: app/validator_tx_fee.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// checkTxFeeWithValidatorMinGasPrices implements the default fee logic, where the minimum price per
+// unit of gas is fixed and set by each validator, can the tx priority is computed from the gas price.
```
Comment (yihuang) [inline app/validator_tx_fee.go:11]:
```suggestion
// unit of gas is fixed and set by each validator, and the tx priority is computed from the gas price.
```

**Pull request merged**